src/aaz_dev/cli/api/portal.py

```
@@ -0,0 +1,110 @@
+    default=Config.SWAGGER_PATH,
+    required=not Config.SWAGGER_PATH,
+    callback=Config.validate_and_setup_swagger_path,
+    expose_value=False,
+    help="The local path of azure-rest-api-specs repo. Official repo is https://github.com/Azure/azure-rest-api-specs"
+)
+@click.option(
+    "--cli-path", '-c',
+    type=click.Path(file_okay=False, dir_okay=True, writable=True, readable=True, resolve_path=True),
+    default=Config.CLI_PATH,
+    required=not Config.CLI_PATH,
+    callback=Config.validate_and_setup_cli_path,
+    expose_value=False,
+    help="The local path of azure-cli repo. Official repo is https://github.com/Azure/azure-cli"
+)
+@click.option(
+    "--cli-extension-path", '-e',
+    type=click.Path(file_okay=False, dir_okay=True, writable=True, readable=True, resolve_path=True),
+    default=Config.CLI_EXTENSION_PATH,
+    required=not Config.CLI_EXTENSION_PATH,
+    callback=Config.validate_and_setup_cli_extension_path,
+    expose_value=False,
+    help="The local path of azure-cli-extension repo. Official repo is https://github.com/Azure/azure-cli-extensions"
+)
+@click.option(
+    "--module", '-m',
+    required=True,
+    help="Name of the module to generate."
+)
+def generate_module_command_portal(module):
+    from cli.controller.az_module_manager import AzMainManager, AzExtensionManager
+    from command.controller.specs_manager import AAZSpecsManager
+    az_main_manager = AzMainManager()
+    az_ext_manager = AzExtensionManager()
+    if az_main_manager.has_module(module):
+        cli_module = az_main_manager.load_module(module)
+        print("Input module: {0} in cli".format(module))
+    elif az_ext_manager.has_module(module):
+        cli_module = az_ext_manager.load_module(module)
+        print("Input module: {0} in cli extension".format(module))
+    else:
+        print("Invalid input module: {0}, please check".format(module))
+        return
```

Raise error instead of `print+return`